pkg/3scale/amp/operator/backend_reconciler.go

```diff
@@ -154,6 +154,30 @@ func (r *BackendReconciler) Reconcile() (reconcile.Result, error) {
 	if err != nil {
 		return reconcile.Result{}, err
 	}
+	if redisStorageUrl != redisQueuesUrl {
+		err = r.ReconcileHpa(component.DefaultHpa(component.BackendListenerName, r.apiManager.Namespace), reconcilers.CreateOnlyMutator)
+		if err != nil {
+			return reconcile.Result{}, err
+		}
+		err = r.ReconcileHpa(component.DefaultHpa(component.BackendWorkerName, r.apiManager.Namespace), reconcilers.CreateOnlyMutator)
+		if err != nil {
+			return reconcile.Result{}, err
+		}
+	} else {
+		// set status message if logical redis db are detected in the backend
+		if r.apiManager.Spec.Backend.ListenerSpec.Hpa || r.apiManager.Spec.Backend.WorkerSpec.Hpa {
+			message := "logical redis instances found in the backend, which is blocking redis async mode, horizontal pod autoscaling for backend cannot be enabled without async mode"
+			r.logger.Info(message)
+			// TODO update the status once the status update framework is in place.
+		}
+	}
+	if redisStorageUrl == redisQueuesUrl {
```

Comment:
WDYT if we move the inside if to above "else" statement? Also, not sure if we even need to log this tbh

Reply:
will leave it for now and drop it in 10499 as we handle the status update there